lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_esm_encode_decode.cpp

```diff
@@ -0,0 +1,299 @@
+    EXPECT_EQ(original_msg.messagetype, decoded_msg.messagetype);              \
+  } while (0)
+
+#define DESTROY_PCO()                                                          \
+  do {                                                                         \
+    bdestroy_wrapper(&original_msg.protocolconfigurationoptions                \
+                          .protocol_or_container_ids[0]                        \
+                          .contents);                                          \
+    bdestroy_wrapper(&original_msg.protocolconfigurationoptions                \
+                          .protocol_or_container_ids[1]                        \
+                          .contents);                                          \
+    bdestroy_wrapper(                                                          \
+        &decoded_msg.protocolconfigurationoptions.protocol_or_container_ids[0] \
+             .contents);                                                       \
+    bdestroy_wrapper(                                                          \
+        &decoded_msg.protocolconfigurationoptions.protocol_or_container_ids[1] \
+             .contents);                                                       \
+  } while (0)
+
+class ESMEncodeDecodeTest : public ::testing::Test {
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+ protected:
+  void fill_pco(protocol_configuration_options_t* pco) {
+    pco->num_protocol_or_container_id    = 2;
+    pco->protocol_or_container_ids[0].id = PCO_CI_P_CSCF_IPV6_ADDRESS_REQUEST;
+    bstring test_string1                 = bfromcstr("teststring");
+    pco->protocol_or_container_ids[0].contents = test_string1;
+    pco->protocol_or_container_ids[0].length   = blength(test_string1);
+    pco->protocol_or_container_ids[1].id =
+        PCO_CI_DSMIPV6_IPV4_HOME_AGENT_ADDRESS;
+    bstring test_string2 = bfromcstr("longer.test.string");
+    pco->protocol_or_container_ids[1].contents = test_string2;
+    pco->protocol_or_container_ids[1].length   = blength(test_string2);
+    return;
+  }
+
+  uint8_t buffer[BUFFER_LEN];
+};
+
+TEST_F(ESMEncodeDecodeTest, TestActivateDedicatedEpsBearerContextAccept) {
+  activate_dedicated_eps_bearer_context_accept_msg original_msg = {0};
+  activate_dedicated_eps_bearer_context_accept_msg decoded_msg  = {0};
+  FILL_COMMON_MANDATORY_DEFAULTS(original_msg);
+  original_msg.presencemask =
+      ACTIVATE_DEDICATED_EPS_BEARER_CONTEXT_ACCEPT_PROTOCOL_CONFIGURATION_OPTIONS_PRESENT;
+  fill_pco(&original_msg.protocolconfigurationoptions);
+
+  int encoded = encode_activate_dedicated_eps_bearer_context_accept(
+      &original_msg, buffer, BUFFER_LEN);
+  int decoded = decode_activate_dedicated_eps_bearer_context_accept(
+      &decoded_msg, buffer, encoded);
+
+  EXPECT_EQ(encoded, decoded);
+  // Header needs to be decoded separately and then can be compared
+  // COMPARE_COMMON_MANDATORY_DEFAULTS();
```

Comment:
Remove commented code

Reply:
I left it on purpose as TODO. The macro will be used when header decoding is added to the unit test.